---
title: Spotify Analysis
subtitle: Series, DataFrame, Rows, Columns, Filtering
date: 2025-03-02
from: markdown+emoji
---

# Spotify Analysis

In [2]:
import pandas as pd
from google.colab import data_table
data_table.enable_dataframe_formatter()
spotify = pd.read_csv('https://bcdanl.github.io/data/spotify_all.csv')
spotify

,pid,playlist_name,pos,artist_name,track_name,duration_ms,album_name
0,0,Throwbacks,0,Missy Elliott,Lose Control (feat. Ciara & Fat Man Scoop),226863,The Cookbook
1,0,Throwbacks,1,Britney Spears,Toxic,198800,In The Zone
2,0,Throwbacks,2,Beyoncé,Crazy In Love,235933,Dangerously In Love (Alben für die Ewigkeit)
3,0,Throwbacks,3,Justin Timberlake,Rock Your Body,267266,Justified
4,0,Throwbacks,4,Shaggy,It Wasn't Me,227600,Hot Shot
...,...,...,...,...,...,...,...
198000,999998,✝️,6,Chris Tomlin,Waterfall,209573,Love Ran Red
198001,999998,✝️,7,Chris Tomlin,The Roar,220106,Love Ran Red
198002,999998,✝️,8,Crowder,Lift Your Head Weary Sinner (Chains),224666,Neon Steeple
198003,999998,✝️,9,Chris Tomlin,We Fall Down,280960,How Great Is Our God: The Essential Collection


## Favorite Artist
* my favorite artist is dire straits
* I will be analyzing their info from spotify

In [7]:
spotify_ds = spotify[spotify['artist_name'] == 'Dire Straits']
spotify_ds

,pid,playlist_name,pos,artist_name,track_name,duration_ms,album_name
22205,354,Cruisin,0,Dire Straits,Walk Of Life,249960,Brothers In Arms
40725,611,Oldies,32,Dire Straits,Money For Nothing,510933,Brothers In Arms
42566,637,Classic,120,Dire Straits,Sultans Of Swing,348624,Dire Straits
46747,694,PARTY,77,Dire Straits,Sultans Of Swing,348624,Dire Straits
47872,712,classic,1,Dire Straits,Money For Nothing,510933,Brothers In Arms
47919,712,classic,48,Dire Straits,Walk Of Life,249960,Brothers In Arms
47924,712,classic,53,Dire Straits,Sultans Of Swing,348624,Dire Straits
50923,756,kitchen,10,Dire Straits,Romeo And Juliet,360666,Making Movies
53575,804,Good Stuff,36,Dire Straits,Sultans Of Swing,348624,Dire Straits
57280,849,chill,72,Dire Straits,Sultans Of Swing,348624,Dire Straits


### Playlists
* Dire Straits appear in 26 different playlists in this dataframe
* they appear the most in the "Classic Rock" data frame
* on average dire straits are added at the 54th spot on people's playlists

In [12]:
spotify_ds['playlist_name'].nunique()

26

In [45]:
spotify['pos'].mean()

54.39170728011919

In [26]:
playlist_count = spotify_ds['playlist_name'].value_counts()
playlist_count

,count
playlist_name,
Classic Rock,7
classic,3
Rock,3
mom!,2
Cruzin,2
Driving,2
Cruisin,1
80s music,1
car,1


### Albums
* three dire straits albums appear in this dataframe
* Brothers in arms being the most popular
* Making movies being the least popular

In [22]:
album_count = spotify_ds['album_name'].value_counts()
album_count

,count
playlist_name,
Classic Rock,7
classic,3
Rock,3
mom!,2
Cruzin,2
Driving,2
Cruisin,1
80s music,1
car,1


### Songs
* Sultans of swing is the most popular
* Why worry is the least popular

In [19]:
song_count = spotify_ds['track_name'].value_counts()
song_count

,count
track_name,
Sultans Of Swing,14
Money For Nothing,8
Walk Of Life,7
Romeo And Juliet,5
Ride Across The River,1
The Man's Too Strong,1
Brothers In Arms,1
One World,1
Why Worry?,1


## Popular Artists
the top 5 most popular artists are
1. drake
2. Kanye West
3. Kendrick Lamar
4. Rihanna
5. The Weeknd

In [29]:
artist = spotify['artist_name'].value_counts()
artist.head(5)

,count
artist_name,
Drake,2715
Kanye West,1065
Kendrick Lamar,1035
Rihanna,915
The Weeknd,913


### Most Popular Artist (Drake)
* most popular song is one One Dance
  * this song appears in 2715 playlists

In [36]:
drake = spotify[spotify['artist_name']=="Drake"]
song_count_drake = drake['track_name'].value_counts()
song_count_drake

,count
track_name,
One Dance,143
Jumpman,95
Fake Love,92
Hotline Bling,83
Passionfruit,72
...,...
Karaoke,2
Buried Alive Interlude,2
The Calm,1


In [40]:
pop_drake_song = drake.set_index('track_name')
pop_drake_song.loc['One Dance', ["playlist_name"]]

,playlist_name
track_name,
One Dance,boat
One Dance,party people
One Dance,w o r k o u t
One Dance,English
One Dance,Summer Chill
...,...
One Dance,mixx
One Dance,(Guilty)
One Dance,Marathon


In [41]:
pop_drake_song.shape[0]

2715

## Popular Songs
most popular songs are:
1. closer
2. one dance
3. HUMBLE
4. home
5. roses

In [44]:
songs = spotify['track_name'].value_counts()
songs.head(5)

,count
track_name,
Closer,205
One Dance,144
HUMBLE.,142
Home,136
Roses,132
